app/src/main/java/com/acsbendi/requestinspectorwebview/RequestInspectorJavaScriptInterface.kt

```diff
@@ -34,7 +35,20 @@ internal class RequestInspectorJavaScriptInterface(webView: WebView) {
         val headers: Map<String, String>,
         val trace: String,
         val enctype: String?
-    )
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (other !is RecordedRequest) return super.equals(other)
+            return url == other.url
+                    && method == other.method
+                    && body == other.body
+                    && headers == other.headers
+                    && formParameters == other.formParameters
+        }
+
+        override fun hashCode(): Int {
+            return super.hashCode()
+        }
```

Comment:
I don't think these are needed, Kotlin generates `equals` and `hashCode` for data classes.

Reply:
I presume the idea is to make the two requests equal only when a subset of parameters are equal, but for example not necessarily "trace" or "enctype".